support/k8s/k8s-workload-registrar/mode-crd/webhook/webhook_svid.go

```diff
@@ -0,0 +1,180 @@
+	expiresAt time.Time
+}
+
+// NewSVID sets up the data structure and mints a new SVID for the webhok
+func NewSVID(ctx context.Context, config SVIDConfig) (*SVID, error) {
+	err := os.MkdirAll(config.WebhookCertDir, certDirMode)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := spiffeid.New(config.TrustDomain, "k8s-workload-registrar", config.Cluster, "webhook")
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate Spiffe ID: %w", err)
+	}
+
+	return &SVID{
+		c:  config,
+		id: id,
+	}, nil
+}
+
+// MintSVID requests the SPIRE Server to mint a new SVID for the webhook
+func (e *SVID) MintSVID(ctx context.Context, key crypto.Signer) (err error) {
+	e.c.Log.Debug("Minting new SVID for webhook")
+	// Generate key if not passed in
+	if key == nil {
+		key, err = ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+		if err != nil {
+			return fmt.Errorf("unable to generate Key: %w", err)
+		}
+	}
+
+	// Generate Certificate Signing Request
+	csr, err := x509.CreateCertificateRequest(rand.Reader, &x509.CertificateRequest{
+		URIs: []*url.URL{e.id.URL()},
+		DNSNames: []string{
+			e.c.WebhookServiceName + "." + e.c.Namespace + ".svc",
+		},
+	}, key)
+	if err != nil {
+		return fmt.Errorf("unable to generate CSR: %w", err)
+	}
+
+	// Mint new SVID
+	resp, err := e.c.S.MintX509SVID(ctx, &svidv1.MintX509SVIDRequest{
+		Csr: csr,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to make Mint SVID Request: %w", err)
+	}
+	e.beginsAt = time.Now()
+	e.expiresAt = time.Unix(resp.Svid.GetExpiresAt(), 0)
+
+	// Write SVID and key to disk
+	return e.dumpSVID(resp.Svid, key)
+}
+
+// SVIDRotator requests a new certificate when half of its lifetime is left
+func (e *SVID) SVIDRotator(ctx context.Context) error {
+	for {
+		lifetime := e.expiresAt.Sub(e.beginsAt) / 2
+		ttl := time.Until(e.expiresAt)
+		select {
+		case <-time.After(ttl - lifetime):
+			if err := e.MintSVID(ctx, nil); err != nil {
+				return fmt.Errorf("unable to mint SVID: %w", err)
+			}
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+
+// dumpBundles takes a X509SVIDResponse, representing a svid message from the SPIRE Server
+// and write the certs to disk
+func (e *SVID) dumpSVID(svid *spiretypes.X509SVID, key crypto.Signer) error {
+	// Convert certificates to PEM
+	svidPEM := new(bytes.Buffer)
+	for _, certDER := range svid.CertChain {
+		_ = pem.Encode(svidPEM, &pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: certDER,
+		})
+	}
+
+	// Convert key to PEM
+	keyBytes, err := x509.MarshalPKCS8PrivateKey(key)
+	if err != nil {
+		return err
+	}
+	keyPEM := new(bytes.Buffer)
+	_ = pem.Encode(keyPEM, &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: keyBytes,
+	})
+
+	// Write certificates to disk
+	certsFileName := path.Join(e.c.WebhookCertDir, certsFileName)
+	certsFile, err := os.OpenFile(certsFileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, certsFileMode)
+	if err != nil {
+		return err
+	}
+	_, err = svidPEM.WriteTo(certsFile)
+	if err != nil {
+		return err
+	}
```

Comment:
These files (both cert and key) aren't being closed. However, in this case, you can just use `os.WriteFile`...
```suggestion
	if err := os.WriteFile(certsFileName, svidPEM.Bytes(), certsFileMode); err != nil {
		return err
	}
```